plugins/outputs/sematext/sematext.go

```diff
@@ -151,11 +151,20 @@ func (s *Sematext) Write(metrics []telegraf.Metric) error {
 		s.Log.Debugf("Sending metrics to %s response status code: %d", s.metricsURL, res.StatusCode)
 
 		success := res.StatusCode >= 200 && res.StatusCode < 300
+		badRequest := res.StatusCode >= 400 && res.StatusCode < 500
+
 		if !success {
-			// TODO in the future, consider handling the retries for bad-request cases
-			// badRequest := res.StatusCode >= 400 && res.StatusCode < 500
-			// if !badRequest {
-			return s.logAndCreateError(res)
+			errorMsg := fmt.Sprintf("received %d status code, message = '%s' while sending to %s",
+				res.StatusCode, res.Status, s.metricsURL)
+
+			if badRequest {
+				// shouldn't be re-sent as bad request will continue to be a bad request
+				s.Log.Error(errorMsg + " - request will be dropped")
```

Comment:
Does the logger has the `Errorf` or similar method so you can interpolate strings inside the message?